### ISSUE TITLE
Nullable<bool> remains null?

### ISSUE THREAD

**Issue by isolaol** [adamabdelhamed/PowerArgs#74]
Hi Adam,

It looks like the setter of the OptionalFlag property gets called but the value passed to it stays null.

The definition of the command line interface is:
```csharp
class NullableArgs {
	public Nullable<bool> OptionalFlag{get;set;}
}
```
and the use of the inteface is here:
```csharp
NullableArgs parsed;

parsed = Args.Parse<NullableArgs>("-OptionalFlag");
```

Kind regards,
Olli-Pekka Isola

**Comment by adamabdelhamed** [comment]
I think booleans get treated differently.  And why would you need a nullable bool?  It will be false if not specified, true if specified. Here are some examples.

"-OptionalFlag" > resolves to true
"" > resolves to false
"-OptionalFlag false" > resolves to false

**Comment by isolaol** [comment]
Sorry about improper reasoning! Let me try again. The use case for nullable boolean is this:

The program has a single class for options wether they come from registry, configuration file or command line. PowerArgs is used to populate the option class with values from command line. some other methods are called to populate option class from XML file, from registry and so on. The class has a method '''C# Merge(Options other) ''' to combine multiple instance (one from each source) as one.

The workflow on the main procedure goes as follows:

1. Application defaults are read from one source (eg. registry) as a first instance of option class.
2. User preferences or last settings are read from second source (eg. configuration file in user documents/profile) as a second instance of option class. This instance is then merged with the first instance by calling Merge() of the first instance. User preferences override application defaults but only those settings that are specified in user preferences.
3. Command line options are read as a third instance of the options class. This instance is then merged with the first instance by calling Merge() of the first instance. Command line options override values from two previous steps but only those settings that are specified in command line.

Nullable boolean (and other nullable types as well) comes handy in this scenario since it helps to decide whether to override previously acquired value or not. So, the interpretation the of nullable boolean is:

Null: not specified, do not overwrite previous value
False: specified as False, overwrite
True: specified as True, overwrite

**Comment by adamabdelhamed** [comment]
Sorry it took me months to answer this, but I don't think I'm going to make a change here.  Your scenario is very unusual, and I'd rather not make risky changes here to accommodate it.  While I'm sure nullable support for bools would make your merge easier I'm also sure you can do it without it.  But false meaning 'not specified' is a core element of boolean arguments.  I don't want to change that meaning.
